torch/_inductor/graph.py

```diff
@@ -505,8 +549,13 @@ def run_node(self, n: torch.fx.Node):
                 # requiring a stride order for a non-dense output wouldn't
                 # recreate the same strides, and would fail with view, defer for now.
                 if dense and len(strides):
+                    stride_order = ir.get_stride_order(strides)
+
+                    if len(result.get_size()) == 4 and (n in self.nodes_prefer_channels_last):
+                        stride_order = ir.NHWC_STRIDE_ORDER
+
                     result = ir.ExternKernel.require_stride_order(
-                        result, ir.get_stride_order(strides)
+                        result, stride_order
```

Comment:
@XiaobingSuper  yep, I reverted 2 lines in aot_autograd.py temporarily to achieve that. Will need figure out the correct solution.

Reply:
I'm still hesitate if we should enforce the same stride as eager for output. Even if we change stride, user still get tensor with exactly the same shape and content. @Chillee mentioned that one caveat is the previous success .view() may fail. But we could use .reshape() instead.

If we really want to do this, we need consider inference and training differently. For inference, all graph outputs are user visible, so we should enforce eager stride for all graph outputs.

For training, things are different. The fwd graph returns saved activations and fwd pass outputs; the backward graph returns gradients for model parameters. 
1. The compiler can freely decide activations stride since they are not user visible
2. Enforce eager stride for fwd pass outputs should be fine since there are usually not a lot of them
3. should we enforce eager stride for gradients of model parameters then? Doing this may have perf impact.

I can follow up on this if people have strong opinions.